shared/index.d.ts

```diff
@@ -989,12 +1071,74 @@ declare interface SettingsSlider extends Panel {
 	RestoreCVarDefault(): void;
 }
 
+declare interface SettingsKeyBinder extends Panel {
+	bind: string;
+
+	OnShow(): void;
+}
+
+declare interface SettingsToggle extends Panel {
+	convar: string;
+
+	/** Set the cvar back to default */
+	RestoreCVarDefault(): void;
+
+	OnShow(): void;
+}
+
+declare interface SettingsEnum extends Panel {
+	convar: string;
+}
+
+declare interface SettingsEnumDropDown extends Panel {
+	convar: string;
+
+	OnShow(): void;
+
+	RefreshDisplay(): void;
+
+	/** Set the cvar back to default */
+	RestoreCVarDefault(): void;
+}
+
 declare interface AvatarImage extends Panel {
 	accountid: string;
 	
 	steamid: string;
 }
 
+declare interface BaseBlurTarget extends Panel {
+	/**
+	 * Add a panel to the blur list
+	 * @param panel 
+	 */
+	AddBlurTarget(panel: Panel);
+
+	/**
+	 * Remove a panel from the blur list
+	 * @param panel 
+	 */
+	RemoveBlurPanel(panel: Panel);
+}
+
+declare interface TripleMonitorBackground extends Panel {
+}
+
+interface ClockTypeEnum {
+	NONE: 'none',
+	REALTIME: 'realtime',
+	WALL: 'wall',
+	GAME: 'game',
+	GAMETICK: 'game-tick',
+	GAMESERVER: 'game-server'
+}
+
+declare interface CountdownTimer extends Panel {
+	timeleft: number;
+	
+	clocktype: ClockTypeEnum;
```

Comment:
With the current types, this will make it assume that `clocktype` has all of the properties of the `ClockTypeEnum` interface. This should be changed to `ClockType` if `ClockTypeEnum` is kept as an interface rather than a typescript enum.